### PR TITLE
Promote allowUnfreePredicate in remediation message

### PR DESCRIPTION
**Pull request by raboof**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/stdenv/generic/check-meta.nix [hunk 1]
```diff
@@ -80,14 +80,41 @@ let
...
+      Or configure a predicate to whitelist specific packages:
+        { nixpkgs.config.allowUnfreePredicate = pkg: builtins.elem (lib.getName pkg) [
+            "package-name-to-whitelist"
+          ];
+        }
+
+      in configuration.nix to override this.
+
+      b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
+        { allowUnfree = true; }
```
Comment (wamserma) [inline pkgs/stdenv/generic/check-meta.nix:105]: or by setting the environment-variable `NIXPKGS_ALLOW_UNFREE=1`, which especially useful when unfree packages are required only occasionally with `nix search`, `nix run` or `nix-shell`.

**Review by wamserma — APPROVED** [review]

File: pkgs/stdenv/generic/check-meta.nix [hunk 2]
```diff
@@ -80,14 +80,44 @@ let
...
     unknown-meta = x: "";
   };
+  remediate_unfree = attrs:
+    ''
+      a) For `nixos-rebuild` you can allow unfree packages globally:
+        { nixpkgs.config.allowUnfree = true; }
+
+      Or configure a predicate to whitelist specific packages:
+        { nixpkgs.config.allowUnfreePredicate = pkg: builtins.elem (lib.getName pkg) [
+            "package-name-to-whitelist"
```
Comment (matthewbauer) [inline pkgs/stdenv/generic/check-meta.nix:98]: we could actually include the package name here, i think. `attrs.name`

Comment (raboof) [reply pkgs/stdenv/generic/check-meta.nix:98]: Great idea, done!

**Review by timokau — APPROVED** [review]
Good idea, thanks!

**Pull request merged**